### PR TITLE
Viewing party

### PR DESCRIPTION
**Pull request by janegreene**
Created viewing party function where a user can invite friends to watch a movie.
Once the event is created it appears on the dashboard of the host user and the invited user.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -21,3 +21,6 @@
 
 # Ignore application configuration
 /config/application.yml
+
+# Ignore Google Calendar API credentials
+credentials.json
```
Comment (janegreene) [inline .gitignore:26]: For use with google calendar API

File: app/models/party_guest.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class PartyGuest < ApplicationRecord
+  belongs_to :view_party
+  belongs_to :user
+
+  def self.create_invites(party, friends)
+    friends.delete("")
```
Comment (janegreene) [inline app/models/party_guest.rb:6]: Comes over as an array with empty strings.

File: app/views/dashboard/show.html.erb [hunk 3]
```diff
@@ -21,5 +21,23 @@
 <br>
 <section class='viewing-parties'>
   <p>Viewing Parties</p>
+<% if !current_user.view_parties.empty? %>
```
Comment (janegreene) [inline app/views/dashboard/show.html.erb:24]: Parties the user is hosting

File: app/views/dashboard/show.html.erb [hunk 4]
```diff
@@ -21,5 +21,23 @@
...
+  <% current_user.view_parties.each do |party| %>
+  <div class="party-<%=party.id%>">
+    <%= party.title %>
+    <%= party.date %>
+    <%= party.time %>
+    <%= image_tag party.poster %>
+  </div>
+  <% end %>
+<% end %>
+<% if !current_user.party_guests.empty? %>
```
Comment (janegreene) [inline app/views/dashboard/show.html.erb:34]: Parties the user is invited to

File: calendar/quickstart/quickstart.rb [hunk 5]
```diff
@@ -0,0 +1,57 @@
+require "google/apis/calendar_v3"
```
Comment (janegreene) [inline calendar/quickstart/quickstart.rb:1]: Not sure if we need this after speaking with Dione

File: db/schema.rb [hunk 6]
```diff
@@ -32,5 +41,20 @@
...
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.index ["user_id"], name: "index_view_parties_on_user_id"
+  end
+
   add_foreign_key "friendships", "users"
+  add_foreign_key "party_guests", "users"
+  add_foreign_key "party_guests", "view_parties"
+  add_foreign_key "view_parties", "users"
 end
```
Comment (michaeljevans) [inline db/schema.rb:60]: renaming these resources and updating file names/method calls/object instantiations was painful

File: spec/features/viewing_parties/new_spec.rb [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+
+    visit dashboard_path
+    within '.viewing-parties' do
+      expect(page).to have_content('Jack Reacher')
+      expect(page).to have_content('2020-12-08')
+      expect(page).to have_content('21:00')
+      expect(page).to have_css("img[src*='https://image.tmdb.org/t/p/w185/7baSUtFKi8PQ9SLo6ECYBfAW2K8.jpg']")
+    end
+  end
+end
```
Comment (michaeljevans) [inline spec/features/viewing_parties/new_spec.rb:64]: this is a nice test :)

File: calendar/quickstart/quickstart.rb (hunk 5 above)
Comment (michaeljevans) [reply calendar/quickstart/quickstart.rb:1]: agreed. it seems like we are overcomplicating things

**Pull request merged**